packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import { act, render, screen } from "@testing-library/react"
+import React from "react"
+import userEvent from "@testing-library/user-event"
+import { DatePicker } from "./DatePicker"
+import "@testing-library/jest-dom"
+
+const defaultProps = {
+  id: "date-picker",
+  labelText: "Choose date",
+  selectedDate: new Date(2022, 2, 1),
+  initialMonth: new Date(2022, 2),
+  onDayChange: () => jest.fn(),
+}
+
+describe("<DatePicker />", () => {
+  it("renders DatePicker and shows/hides calendar on focus of input", async () => {
+    render(<DatePicker {...defaultProps} />)
+
+    const element = screen.getByDisplayValue("Mar 1, 2022")
+
+    // Make sure calendar popup is not in the DOM
+    expect(screen.queryByRole("dialog")).not.toBeInTheDocument()
+
+    // Focus input and test calendar popup is showing
+    await act(async () => element.focus())
+    expect(element).toHaveFocus()
+    expect(screen.getByRole("dialog")).toBeVisible()
+  })
+
```

Comment:
Could also potentially add a test for focus out here too if you wanted to cover both bases :)